docs/design/pulumi-yaml.md

```diff
@@ -0,0 +1,62 @@
+# Integration with Pulumi YAML Programs
+
+## Summary
+
+This design is describes the approach to implementing the ability to execute Pulumi YAML programs from the Operator, by declaring the program as a Kubernetes resource.
+
+## Motivation
+
+Pulumi YAML is intended as an accessible and easy way to write Pulumi programs. 
+YAML is already the standard for creating Kubernetes resources.
+
+Therefore, combining the two means we can provide an easy way to create projects within the Operator, without requiring the use of an external repo for storing a Pulumi project. Projects can be created, edited, and deleted within Kubernetes, making use of kubectl or other automation, as with any other Kubernetes resource.
+
+## Design
+
+Add a new CRD to the Operator, specifying a Pulumi program. This will be [Pulumi YAML](https://www.pulumi.com/docs/reference/yaml/) (as used with the Pulumi CLI) under a 'Program' field, in addition to the usual, compulsory Kubernetes API fields. Additionally add a field to the Stack resource referencing the program, as an alternative to projectRepo (or upcoming Flux fields).
```

Comment:
```suggestion
Add a new CRD to the operator API, specifying a Pulumi program. This will be [Pulumi YAML](https://www.pulumi.com/docs/reference/yaml/) (as used with the Pulumi CLI) under a 'Program' field, in addition to the usual, compulsory Kubernetes API fields. Additionally add a field to the Stack resource referencing the program, as an alternative to projectRepo (or upcoming Flux fields).
```